tests/auth/ldap_test.py

```diff
@@ -6,33 +6,92 @@
 """
 import pytest
 
+# module must have ldap.INVALID_CREDENTIALS
+INVALID_CREDENTIALS = Exception
```

Comment:
You need a unique test here or the use of INVALID_CREDENTIALS will match any exception, changing the semantics of the code. Change the comment to use Mock and move this down to where initialize is so all the mock stuff is together.

```py
class INVALID_CREDENTIALS():
    pass
```